tracer/test/test-applications/integrations/Samples.DataStreams.RabbitMQ/Program.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Text;
+using RabbitMQ.Client;
+
+namespace Samples.DataStreams.RabbitMQ
+{
+    public static class Program
+    {
+        private static readonly string Queue = nameof(Queue);
+        private static readonly string Exchange = nameof(Exchange);
+        private static readonly string RoutingKey = nameof(RoutingKey);
+        private static readonly string Message = nameof(Message);
+        private static readonly string Host = Environment.GetEnvironmentVariable("RABBITMQ_HOST") ?? "localhost";
+        
+        public static void Main(string[] args)
+        {
+            var factory = new ConnectionFactory() { HostName = Host };
+            using var connection = factory.CreateConnection();
+            using var model = connection.CreateModel();
+            
+            // produce/consume operation
+            PublishMessageToQueue(model, Queue, Message);
+            var msg = GetMessage(model, Queue);
+
+            // continuing the chain, now though an exchange
+            var queueName = PublishMessageToExchange(model, Exchange, RoutingKey, msg);
+            GetMessage(model, queueName);
+        }
```

Comment:
I have a few concerns with this setup and how it maps to the DSM values:

- We're not creating a "parent" trace anywhere here, which means most of these spans are disconnected. e.g. the "queue declare" trace is separate from the "get/publish" trace. The "get/publish" trace _does_ include two spans in the trace, but the span is immediately opened and closed-there's no continuing context.
- "continuing the chain, now though an exchange". Except, we're not. There's no chain continuation, because there's no ongoing trace, so there's no context passed on.
- People sometimes look at these samples to understand how things _should_ work, and this is, unfortunately, not canonical, because of the way we lose the trace/dsm context after the GET. I think we need to look into it
- There's no support for "manual" extraction of the trace context here, which _currently_ is the "correct" way to use RabbitMQ with the tracer. If support for that is coming in a later PR, that's fine, but we need to consider it before we can consider rabbitMQ to be supported fully I think

Reply:
Please, take a look at changes inside DataStreamsManager. This is the DSM context propagation which is now inline with what we have for Java tracer. The direction check there makes sure that we do not create fake chains (on fan outs, for instance), and the chain resets as soon as you pick any context from the headers.